### PR TITLE
Mychel larson

### PR DESCRIPTION
**Pull request by myke47**

### CODE REVIEW

File: ANSWERS.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+*HOC is an advanced technique for reusing component logic. These are not part of Reacts API but rather patterns that emerge from Reacts compositional nature.  Higher-Order Component is a function that takes a component in as one parameter and returns a new component. HOCs are commonly used in third-party React libraries. If composed properly, this replaces the need to use mixins for cross-cutting between components of an application. 
+
+- [ ] What are three different ways to style components in React? Explain some of the benefits of each.
+
+*Styled-components = a library for writing CSS in our JS files using a tagged template literals, which are essentially invoking functions that pass strings with styles in them. With this method, we can actually pass in props that will impact the color of our elements or components such as a button. 
+
+*Reactstrap = bootstrap modified for React to implement pre-built components that are already styled, from the Reactstrap library. 
+
+*CSS = traditional styling sheet that uses a separate file to controls styles throughout a component or entire application.
```
Comment (Alex-AK) [inline ANSWERS.md:21]: Answers are great.

File: lambdatimes/src/components/Content/Card.js [hunk 2]
```diff
@@ -1,19 +1,29 @@
...
 
 const Card = props => {
   return (
     <div className="card">
-      <div className="headline">{/* headline goes here */}</div>
+      <div className="headline">{props.card.headline}</div>
       <div className="author">
         <div className="img-container">
-          <img src={'' /* image source goes here */} />
+          <img src={props.card.img} />
```
Comment (Alex-AK) [inline lambdatimes/src/components/Content/Card.js:10]: Nice work. Don't forget to include an alt attribute with an image description for your img tags.

File: lambdatimes/src/components/Content/Card.js [hunk 3]
```diff
@@ -1,19 +1,29 @@
...
 // Make sure to include PropTypes.
+Card.PropTypes = {
+  cardData: PropTypes.shape({
+    headline: PropTypes.string,
+    tab: PropTypes.string,
+    img: PropTypes.string,
+    author: PropTypes.string,
+  })
+}
+
```
Comment (Alex-AK) [inline lambdatimes/src/components/Content/Card.js:27]: Good use of prop types.

File: lambdatimes/src/components/Content/Cards.js [hunk 4]
```diff
@@ -1,15 +1,27 @@
...
     <div className="cards-container">
-      {/* Using the cards prop, map over the list creating a 
-          new Card component for each passing the card as the only prop*/}
-    </div>
-  )
-}
+      {
+        props.cards.map(card => (
+            <Card card={card}/>
+            )
```
Comment (Alex-AK) [inline lambdatimes/src/components/Content/Cards.js:11]: Good use of map to return all of your components.

File: lambdatimes/src/components/Content/Content.js [hunk 5]
```diff
@@ -16,12 +16,19 @@ export default class Content extends Component {
     };
   }
 
-  componentDidMount() {
+  componentDidMount = () => {
     // Once the component has mounted, get the data and reflect that data on the state.
+    this.setState({ 
+      tabs: tabData,
+      cards: cardData
+     })
```
Comment (Alex-AK) [inline lambdatimes/src/components/Content/Content.js:24]: Nice use of setState to set multiple data types to state.

File: lambdatimes/src/components/Content/Content.js [hunk 6]
```diff
@@ -16,12 +16,19 @@ export default class Content extends Component {
...
   }
 
   changeSelected = tab => {
     // this function should take in the tab and update the state with the new tab.
+    this.setState({ 
+      selected: tab,
+    })
```
Comment (Alex-AK) [inline lambdatimes/src/components/Content/Content.js:31]: Good!

File: lambdatimes/src/components/Content/Content.js [hunk 7]
```diff
@@ -37,7 +44,14 @@ export default class Content extends Component {
           of the items from cardData. 
         - else, it should only return those cards whose 'tab' matched this.state.selected.
     */
-    return this.state.cards;
+   const filteredCards = this.state.cards.filter((card) => card.tab === this.state.selected );
```
Comment (Alex-AK) [inline lambdatimes/src/components/Content/Content.js:47]: Nice one line filter method! Woooooo!

File: lambdatimes/src/components/Content/Content.js [hunk 8]
```diff
@@ -37,7 +44,14 @@ export default class Content extends Component {
...
+
+   if (this.state.selected === 'all'){
+     return this.state.cards;
+   } 
+   else{
+     return filteredCards;
+   }
```
Comment (Alex-AK) [inline lambdatimes/src/components/Content/Content.js:54]: Nice conditional.

File: lambdatimes/src/components/Content/Tab.js [hunk 9]
```diff
@@ -1,15 +1,18 @@
...
+import PropTypes from 'prop-types'
 
 const Tab = props => {
   /* Using your props, determine if the `tab` prop matches the `selectedTab` prop, 
       if they match, the className should be: 'tab active-tab', 
       if it is not it should just be 'tab'*/
   return (
     <div
-      className={''}
+      className={props.tab === props.selectedTab ? 'tab active-tab' : 'tab'}
```
Comment (Alex-AK) [inline lambdatimes/src/components/Content/Tab.js:10]: Great use of ternary operator to dynamically set the className.

File: lambdatimes/src/components/Content/Tab.js [hunk 10]
```diff
@@ -1,15 +1,18 @@
...
       onClick={() => {
         /* Replace this dummy click handler function with your selectTabHandler function from props 
          you'll need to pass the `tab` in as an argument to this handler. */
+         props.selectTabHandler(props.tab);
+         props.selectedTab(props.tab);
```
Comment (Alex-AK) [inline lambdatimes/src/components/Content/Tab.js:15]: Nicely handled.

File: lambdatimes/src/components/Content/Tabs.js [hunk 11]
```diff
@@ -1,16 +1,29 @@
...
+  console.log(props)
   return (
     <div className="tabs">
       <div className="topics">
         <span className="title">TRENDING TOPICS:</span>
-        {/* map over the tabs provided on your props, create a new Tab component for each one.
-            give the tab component a `selectTabHandler`, the `selectedTab`, and the `tab` itself as props*/}
+            {props.tabs.map(tab => (
+                <Tab
+                key={Date.now() * Math.random()}
```
Comment (Alex-AK) [inline lambdatimes/src/components/Content/Tabs.js:13]: Pulling out all the stops! There is a potential for this to return the same number for the key, but it's so unlikely I would feel comfortable using this.

File: lambdatimes/src/components/Header.js [hunk 12]
```diff
@@ -1,15 +1,47 @@
...
+flex: 8;
+text-align: center;
+color: #000;
+`
+const TempSpan = styled.span`
+text-align: right;
+margin-right: 25px;
+flex: 1;
+`
+
```
Comment (Alex-AK) [inline lambdatimes/src/components/Header.js:37]: Nice use of styled components.

**Review by Alex-AK — APPROVED** [review]
Site looks great and functions as expected. Nice work meeting MVP this week! Check out my individual comments for notes on specific code.

File: lambdatimes/src/components/Content/Tabs.js (hunk 11 above)
Comment (myke47) [reply lambdatimes/src/components/Content/Tabs.js:13]: I toyed with this idea and waffled a bit on using it or not. I have to review the rest of my code in detail but I feel like I missed an opportunity to use this further.

File: lambdatimes/src/components/Header.js (hunk 12 above)
Comment (myke47) [reply lambdatimes/src/components/Header.js:37]: Going back to refactor the app like stretch suggested next.

File: lambdatimes/src/components/Content/Cards.js (hunk 4 above)
Comment (Alex-AK) [reply lambdatimes/src/components/Content/Cards.js:11]: Don't forget to include a key on your components in a map.

**Pull request merged**